toolkit/block-producer-metadata/pallet/src/lib.rs

```diff
@@ -170,6 +171,15 @@ pub mod pallet {
 		QueryKind = OptionQuery,
 	>;
 
+	/// Storage mapping from block producers to their metadata depositor accounts
+	#[pallet::storage]
+	type BlockProducerMetadataOwners<T: Config> = StorageMap<
```

Comment:
we can try to check if using of StorageDoubleMap will be easier here, to keep data in one structure